src/analysis/asymptotics/negligible.lean

```diff
@@ -0,0 +1,219 @@
+/-
+Copyright (c) 2021 Devon Tuma. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Devon Tuma
+-/
+import analysis.asymptotics.asymptotics
+import analysis.asymptotics.specific_asymptotics
+import data.polynomial.eval
+
+/-!
+# Negligible Functions
+
+This file defines a predicate `negligible f` for a function satisfying
+  one of following equivalent definitions (The definition is in terms of the first condition):
+
+* `f` is `O(x ^ c)` for all (or equivalently sufficiently small) integers `c`
+* `f` is `O(p(x)⁻¹)` for all (or equivalently sufficiently large) polynomials `p`
+* `p(x) * f` is bounded for all polynomials `p`
```

Comment:
This seems the most elementary and generalizable definition (and the "main theorem" is trivial from it). Is there a reason for preferring the one you've used?

Reply:
I had tried this sort of definition originally, but I found it was a lot harder to derive most properties of negligible functions (even though it was easier to prove functions are negligible).

It also seemed a lot easier to prove that the `p(x) * f is bounded` definitions imply the `O(x ^ c)` definitions than the converse, since `is_O_of_div_tendsto_nhds` gives a fairly direct path in the one direction (e.g. in `negligible_of_fpow_mul_tendsto_zero`)